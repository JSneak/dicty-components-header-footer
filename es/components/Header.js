function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from "react";
import logo from "../images/logo.png";
import Link from "../styles/Link";
import { headerStyles as styles } from "../styles/headerStyles";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Input from "@material-ui/core/Input";
import InputLabel from "@material-ui/core/InputLabel";
import InputAdornment from "@material-ui/core/InputAdornment";
import FormControl from "@material-ui/core/FormControl";
import Grow from "@material-ui/core/Grow";
import Search from "@material-ui/icons/Search";

// set base URL for homepage
var Home = "http://dictybase.org";

/**
 * The `Header` component that will be displayed
 * on top of navigation bar in every React web application of [dictyBase](http://dictybase.org).
 */

var Header = function (_Component) {
  _inherits(Header, _Component);

  function Header() {
    var _temp, _this, _ret;

    _classCallCheck(this, Header);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.state = {
      isExpanded: false
    }, _this.onClick = function () {
      _this.setState({ isExpanded: !_this.state.isExpanded });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  Header.prototype.render = function render() {
    var _props = this.props,
        classes = _props.classes,
        _props$home = _props.home,
        home = _props$home === undefined ? Home : _props$home,
        children = _props.children,
        items = _props.items;
    var isExpanded = this.state.isExpanded;

    return React.createElement(
      Grid,
      { container: true, spacing: 8, className: classes.root },
      React.createElement(
        Grid,
        {
          item: true,
          xs: 12,
          sm: 5,
          md: 3,
          lg: 3,
          className: classes.logoContainer },
        React.createElement(
          Link,
          { href: home },
          React.createElement("img", { src: logo, alt: "dictyBase logo", className: classes.image })
        )
      ),
      !isExpanded && React.createElement(
        Grid,
        {
          item: true,
          xs: 12,
          sm: 7,
          md: 5,
          lg: 5,
          className: classes.searchContainer },
        React.createElement(
          FormControl,
          { className: classes.textField },
          React.createElement(
            InputLabel,
            { htmlFor: "guided-search" },
            "Guided Search"
          ),
          React.createElement(Input, {
            id: "input-with-icon-adornment",
            onClick: this.onClick,
            endAdornment: React.createElement(
              InputAdornment,
              { position: "end" },
              React.createElement(Search, null)
            )
          })
        )
      ),
      isExpanded && React.createElement(
        Grow,
        { "in": isExpanded, timeout: 500, style: { opacity: 1 } },
        React.createElement(
          Grid,
          {
            item: true,
            xs: 12,
            sm: 7,
            md: 5,
            lg: 5,
            className: classes.searchContainer },
          React.createElement(
            FormControl,
            { className: classes.textFieldExpanded },
            React.createElement(
              InputLabel,
              { htmlFor: "guided-search" },
              "Guided Search"
            ),
            React.createElement(Input, {
              id: "input-with-icon-adornment",
              autoFocus: true,
              endAdornment: React.createElement(
                InputAdornment,
                { position: "end" },
                React.createElement(Search, null)
              )
            })
          )
        )
      ),
      React.createElement(
        Grid,
        {
          item: true,
          xs: 12,
          sm: 12,
          md: 4,
          lg: 4,
          className: classes.linkContainer },
        children(items)
      )
    );
  };

  return Header;
}(Component);

export default withStyles(styles)(Header);