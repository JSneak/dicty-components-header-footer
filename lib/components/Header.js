"use strict";

exports.__esModule = true;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _logo = require("../images/logo.png");

var _logo2 = _interopRequireDefault(_logo);

var _Link = require("../styles/Link");

var _Link2 = _interopRequireDefault(_Link);

var _headerStyles = require("../styles/headerStyles");

var _styles = require("@material-ui/core/styles");

var _Grid = require("@material-ui/core/Grid");

var _Grid2 = _interopRequireDefault(_Grid);

var _ExpandedSearch = require("./ExpandedSearch");

var _ExpandedSearch2 = _interopRequireDefault(_ExpandedSearch);

var _NormalSearch = require("./NormalSearch");

var _NormalSearch2 = _interopRequireDefault(_NormalSearch);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// set base URL for homepage
var Home = "/";

/**
 * The `Header` component that will be displayed
 * on top of the navigation bar in every React web application of [dictyBase](http://dictybase.org).
 */

var Header = function Header(props) {
  var _useState = (0, _react.useState)(false),
      isExpanded = _useState[0],
      setIsExpanded = _useState[1];

  var classes = props.classes,
      _props$home = props.home,
      home = _props$home === undefined ? Home : _props$home,
      children = props.children,
      items = props.items;


  var handleClick = function handleClick() {
    setIsExpanded(!isExpanded);
  };

  return _react2.default.createElement(
    _Grid2.default,
    { container: true, className: classes.root },
    _react2.default.createElement(
      _Grid2.default,
      { item: true, xs: 12, sm: 5, md: 3, lg: 3, className: classes.logoContainer },
      _react2.default.createElement(
        _Link2.default,
        { href: home },
        _react2.default.createElement("img", { src: _logo2.default, alt: "dictyBase logo", className: classes.image })
      )
    ),
    isExpanded ? _react2.default.createElement(_ExpandedSearch2.default, { isExpanded: isExpanded }) : _react2.default.createElement(_NormalSearch2.default, { handleClick: handleClick }),
    _react2.default.createElement(
      _Grid2.default,
      {
        item: true,
        xs: 12,
        sm: 12,
        md: 4,
        lg: 4,
        className: classes.linkContainer },
      children(items)
    )
  );
};

exports.default = (0, _styles.withStyles)(_headerStyles.headerStyles)(Header);
module.exports = exports["default"];