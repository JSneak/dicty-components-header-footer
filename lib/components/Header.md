## Default header component

```example
        <Header
          downloads="/downloads"
          info="/information"
          cite="/citation"
        />
```
