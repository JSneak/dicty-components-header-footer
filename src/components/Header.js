// @flow
import React, { useState } from "react"
import logo from "../images/logo.png"
import Link from "../styles/Link"
import { headerStyles as styles } from "../styles/headerStyles"
import { withStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import ExpandedSearch from "./ExpandedSearch"
import NormalSearch from "./NormalSearch"

// set base URL for homepage
const Home = "/"

type ItemType = {
  /** url link */
  url: string,
  /** FontAwesome icon to be displayed */
  icon: string,
  /** description of the link that will be displayed */
  text: string,
}

type Props = {
  /** List of header items, inside an array */
  items: Array<ItemType>,
  /** Link for the logo, that goes to the home page */
  home?: string,
  /** Children react node which renders the links in the header */
  children: any,
  /** Material-UI's classes object for styling */
  classes: Object,
}

/**
 * The `Header` component that will be displayed
 * on top of the navigation bar in every React web application of [dictyBase](http://dictybase.org).
 */

const Header = (props: Props) => {
  const [isExpanded, setIsExpanded] = useState(false)
  const { classes, home = Home, children, items } = props

  const handleClick = () => {
    setIsExpanded(!isExpanded)
  }

  return (
    <Grid container className={classes.root} alignItems="center">
      <Grid item xs={12} sm={5} md={3} lg={3} className={classes.logoContainer}>
        <Link href={home}>
          <img src={logo} alt="dictyBase logo" className={classes.image} />
        </Link>
      </Grid>

      <Grid item className={classes.dcr} xs={12} sm={7} md={5} lg={5}>
        <center>
          <br />
          <em>Dicty Community Resource</em>
        </center>
      </Grid>
      <Grid item xs={12} sm={12} md={4} lg={4}>
        <div className={classes.linkContainer}>{children(items)}</div>
        {isExpanded ? (
          <ExpandedSearch isExpanded={isExpanded} />
        ) : (
          <NormalSearch handleClick={handleClick} />
        )}
      </Grid>
    </Grid>
  )
}

export default withStyles(styles)(Header)
