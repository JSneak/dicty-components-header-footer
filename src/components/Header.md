## Default header component

```jsx
const headerTheme = {
  primary: "#004080",
  secondary: "#001b53",
  text: "#004080",
}

const items = [
  {
    url: "/cite",
    icon: "plus",
    text: "Cite Us",
  },
  {
    url: "/downloads",
    icon: "download",
    text: "Downloads",
  },
  {
    url: "/info",
    icon: "info-circle",
    text: "About dictyBase",
  },
  {
    url: "/login",
    icon: "sign-in",
    text: "Login",
    isRouter: true,
  },
]

const style = {
  marginRight: "8px",
  marginTop: "24px",
}

const HeaderLinks = ({ items }) => {
  return (
    <React.Fragment>
      {items.map((link, i) => (
        <a key={i} href={link.url} style={style}>
          {link.text}
        </a>
      ))}
    </React.Fragment>
  )
}

;<Header items={items} render={HeaderLinks} theme={headerTheme} />
```
