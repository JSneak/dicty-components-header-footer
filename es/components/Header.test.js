import React from "react"
import Header from "./Header"
import renderer from "react-test-renderer"

test("matching a snapshot of Header", () => {
  const component = renderer.create(
    <Header downloads="/downloads" info="/information" cite="/citation" />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
