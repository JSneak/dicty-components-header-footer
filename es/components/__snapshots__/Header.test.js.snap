// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Header 1`] = `
<div
  className="sc-htoDjs bYyUcW"
>
  <div
    className="sc-bxivhb buXlKK"
  >
    <a
      className="sc-bwzfXH hGnGxX"
      href="http://dictybase.org"
    >
      <img
        className="sc-bdVaJa jvBHhC"
        src="test-file-stub"
      />
    </a>
  </div>
  <div
    className="sc-EHOje brHOjA"
  >
    <div
      className="sc-htpNat RVmdF"
    >
      <a
        className="sc-bwzfXH hGnGxX"
        href="/citation"
      >
        <span
          aria-hidden={true}
          className="fa fa-plus"
        />
         Cite Us
      </a>
      <a
        className="sc-bwzfXH hGnGxX"
        href="/downloads"
      >
        <span
          aria-hidden={true}
          className="fa fa-download"
        />
         Downloads
      </a>
      <a
        className="sc-bwzfXH hGnGxX"
        href="/information"
      >
        <span
          aria-hidden={true}
          className="fa fa-info-circle"
        />
         About dictyBase
      </a>
    </div>
    <div
      className="sc-gzVnrw eulfAK"
    >
      <input
        className="sc-bZQynM jNzRMB"
        disabled={true}
        placeholder="Guided Search (coming soon)"
      />
      <div
        className="sc-ifAKCX TXZOw"
      >
        <span
          aria-hidden={true}
          className="fa fa-search"
        />
      </div>
    </div>
  </div>
</div>
`;
